### ISSUE TITLE
Wrapper na model ProteinTransformerLightning

### ISSUE THREAD

**Issue by s26435** [Science-Club-UG-BioAI/Nightingale#23]
Napisz wrapper dla głównego modelu (dziedzicz po `pl.LightningModule`):

1. Konstruktor:
* Argumenty: `aa_vocab_size: int`, `desc_vocab_size: int`, `pad_idx: int`, `d_model: int`, `nhead: int`, `num_encoder_blocks: int`, `num_decoder_blocks: int`, `lr: float`, `weight_decay: float` 
* Co robi: 
   * zapisanie hiperparametrów
   * inicjalizacja modelu `ProteinTransformer(aa_vocab_size, desc_vocab_size,  d_model,  nhead,  num_encoder_blocks,  num_decoder_blocks)`
   * inicjalizacja loss: `self.loss_fn = nn.CrossEntropyLoss(ignore_index=pad_idx)`
2. Forward: 
```
def forward(self, src: torch.Tensor, tgt_in: torch.Tensor) -> torch.Tensor:
        return self.model(src, tgt_in)
```
3. `def _step(self, batch: Tuple[torch.Tensor, torch.Tensor, torch.Tensor], stage: str, ) -> torch.Tensor:`
* co robi:  logogwanie do tensorboard, obliczanie loss
* zwraca loss

4. training_step:
```
def training_step(self, batch, batch_idx):
        return self._step(batch, "train")
```
5. validation_step:
```
def validation_step(self, batch, batch_idx):
        self._step(batch, "val")
```
6. test_step:
```
def test_step(self, batch, batch_idx):
        self._step(batch, "test")
```
7. configure_optimizers:
```
def configure_optimizers(self):
        optimizer = torch.optim.AdamW(
            self.parameters(),
            lr=self.hparams.lr,
            weight_decay=self.hparams.weight_decay,
        )
        scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
            optimizer, mode="min", factor=0.5, patience=3, verbose=True
        )
        return {
            "optimizer": optimizer,
            "lr_scheduler": {
                "scheduler": scheduler,
                "monitor": "val_loss",
                "frequency": 1,
                "interval": "epoch",
            },
        }
```
